### PR TITLE
feat: abstract out common component props as oc base props, add ability to forward ref

### PR DESCRIPTION
**Pull request by ychhabra-eightfold**

### CODE REVIEW

File: src/components/OcBase/Base.types.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import React, { HTMLAttributes } from 'react';
+
+export interface OcBaseProps<T> extends HTMLAttributes<T> {
+    classNames?: string;
```
Comment (dkilgore-eightfold) [inline src/components/OcBase/Base.types.ts:4]: Should this also contain `style`?

**Review by dkilgore-eightfold — CHANGES_REQUESTED** [review]

File: src/components/OcBase/Base.types.ts (hunk 1 above)
Comment (ychhabra-eightfold) [reply src/components/OcBase/Base.types.ts:4]: style is already part of HTMLAttributes

File: src/components/OcBase/Base.types.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import React, { HTMLAttributes } from 'react';
+
+export interface OcBaseProps<T> extends HTMLAttributes<T> {
```
Comment (dkilgore-eightfold) [inline src/components/OcBase/Base.types.ts:3]: @ychhabra-eightfold, come to think of it we may need to discuss this further and include @thiyagaraj-eightfold as he owns the task.

**Review by dkilgore-eightfold — CHANGES_REQUESTED** [review]

File: src/components/OcBase/Base.types.ts (hunk 2 above)
Comment (ychhabra-eightfold) [reply src/components/OcBase/Base.types.ts:3]: yeah completely agree @dkilgore-eightfold, @thiyagaraj-eightfold please share ur proposal. 
Just to be mindfull of timelines i'm currently blocked on this, which is why I wanted to add data-test-id as a prop in every component.

Comment (thiyagaraj-eightfold) [reply src/components/OcBase/Base.types.ts:3]: Hey -it's not really solving the problem of primitives, it's just creating a standard interface for the component. 
Just standardizing the props.
If it's blocking, let's merge it for now to unblock @ychhabra-eightfold

Comment (thiyagaraj-eightfold) [reply src/components/OcBase/Base.types.ts:3]: Hey -it's not really solving the problem of primitives, it's just creating a standard interface for the component. 
Just standardizing the props.
If it's blocking, let's merge it for now to unblock @ychhabra-eightfold

**Review by thiyagaraj-eightfold — APPROVED** [review]

File: src/components/OcBase/Base.types.ts (hunk 2 above)
Comment (ychhabra-eightfold) [reply src/components/OcBase/Base.types.ts:3]: That was never the intention 🙂. I just wanted to add data-test-id everywhere, this was the only sane way of doing it imo. You can still work on creating primitives.

**Pull request merged**